bpf-prog/ipv6-healthprobe-tc/cmd/ipv6-healthprobe-tc/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"net"
+	"os/exec"
+
+	"github.com/Azure/azure-container-networking/bpf-prog/ipv6-healthprobe-tc/pkg/egress"
+	"github.com/Azure/azure-container-networking/bpf-prog/ipv6-healthprobe-tc/pkg/ingress"
+	"github.com/vishvananda/netlink"
+
+	"github.com/cilium/ebpf/rlimit"
+	"go.uber.org/zap"
+)
+
+var logger *zap.Logger
+
+func main() {
+	// Set up logger
+	config := zap.NewProductionConfig()
+	config.OutputPaths = []string{"stdout", "/var/log/ipv6-healthprobe-tc.log"}
+	logger, _ = config.Build()
+
+	// Remove resource limits for kernels <5.11.
+	if err := rlimit.RemoveMemlock(); err != nil {
+		logger.Error("Removing memlock", zap.Error(err))
+		return
+	}
+
+	// Check 'nft -n list tables ip6' to see if table exists
+	cmd := exec.Command("nft", "-n", "list", "tables", "ip6")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		logger.Error("error running 'nft -n list tables ip6'", zap.Error(err), zap.String("output", string(output)))
+		return
+	}
+
+	// if azureSLBProbe table exists, delete it
+	if bytes.Contains(output, []byte("azureSLBProbe")) {
+		cmd := exec.Command("nft", "delete", "table", "ip6", "azureSLBProbe")
+		err = cmd.Run()
+		if err != nil {
+			logger.Error("failed to run 'nft delete table ip6 azureSLBProbe'", zap.Error(err))
+			return
+		}
+	}
+
+	ifname := "eth0"
+	iface, err := net.InterfaceByName(ifname)
+	if err != nil {
+		logger.Error("Getting interface", zap.String("interface", ifname), zap.Error(err))
+	}
+	logger.Info("Interface has index", zap.String("interface", ifname), zap.Int("index", iface.Index))
+
+	// Create a qdisc filter for traffic on the interface.
+	fq := &netlink.GenericQdisc{
+		QdiscAttrs: netlink.QdiscAttrs{
+			LinkIndex: iface.Index,
+			Handle:    netlink.MakeHandle(0xffff, 0),
+			Parent:    netlink.HANDLE_CLSACT,
+		},
+		QdiscType: "clsact",
+	}
+	if err := netlink.QdiscReplace(fq); err != nil {
```

Comment:
does replace deletes and creates filter?

Reply:
yes as long as there is an existing qdisc with the same attributes as this new one, it will be replaced